packages/@aws-cdk/app-staging-synthesizer-alpha/README.md

```diff
@@ -0,0 +1,394 @@
+│                             │││                  │                   │││┌────────────────┐│  │   └───────────┘│
+│                             │││                  │                   ││││ ECR Repository ││  │                │
+│                             │││                  │                   ││││  for Staging   │◀──┘                │
+│                             │││                  │                   ││││     Assets     ││                   │
+│                             │││                  │                   │││└────────────────┘│                   │
+│                             │││                  │                   │││                  │                   │
+│                             │││                  │                   │││                  │                   │
+│                             │││                  │                   │││                  │                   │
+│                             │││                  │                   │││                  │                   │
+│                             │││                  │                   │││                  │                   │
+│                             ││└──────────────────┘                   ││└──────────────────┘                   │
+└─────────────────────────────┘└───────────────────────────────────────┘└───────────────────────────────────────┘
+```
+
+This allows staging resources to be created when needed next to the CDK App. It has the following
+benefits:
+
+- Resources between separate CDK Apps are separated so they can be cleaned up and lifecycle
+controlled individually.
+- Users have a familiar way to customize staging resources in the CDK Application.
+
+> As this library is `experimental`, the accompanying Bootstrap Stack is not yet implemented. To use this
+> library right now, you must reuse roles that have been traditionally bootstrapped.
+
+## Synthesizer
+
+To use this library, supply the `AppStagingSynthesizer` in as the default synthesizer to the app.
+This will ensure that a Staging Stack will be created next to the CDK App to hold the staging resources.
+
+`AppStagingSynthesizer` comes with static methods covering the use-cases for the synthesizer. 
+
+### Using the Default Staging Stack per Environment
+
+The most common use case will be to use the built-in default resources. In this scenario, the
+synthesizer will create a new Staging Stack in each environment the CDK App is deployed to store
+its staging resources. To use this kind of synthesizer, use `AppStagingSynthesizer.defaultResources()`.
+
+```ts
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
+    appId: 'my-app-id',
+  }),
+});
+```
+
+Every CDK App that uses the `DefaultStagingStack` must include an `appId`. This should
+be an identifier unique to the app and is used to differentiate staging resources associated
+with the app.
+
+### Using a Custom Staging Stack per Environment
```

Comment:
Probably put the content about 'escaping from the straitjacket' below the 'customizing the straitjacket' content.